### PR TITLE
update markets with pricing fix, and improve docs

### PR DESCRIPTION
**Pull request by whyrusleeping**
pulling in https://github.com/filecoin-project/go-fil-markets/pull/491 for less nonsensical deal pricing

### CODE REVIEW

File: cli/client.go [hunk 1]
```diff
@@ -299,8 +299,20 @@ var clientLocalCmd = &cli.Command{
...
-	Usage:     "Initialize storage deal with a miner",
+	Name:  "deal",
+	Usage: "Initialize storage deal with a miner",
+	Description: `Make a deal with a miner.
+
+dataCid comes from running 'lotus client import'.
+
+miner is the address of the miner you wish to make a deal with.
+
+price is measured in FIL/GB/Block. Miners usually don't accept a bid
```
Comment (dirkmc) [inline cli/client.go:310]:
```suggestion
price is measured in FIL/GB/Epoch. Miners usually don't accept a bid
```

**Review by dirkmc — APPROVED** [review]
LGTM 👍 

The current version of markets v1.1.8 caused issues when lotus clients on a previous version of markets tried to propose a deal with lotus miners on v1.1.8: https://github.com/filecoin-project/lotus/pull/5608

https://github.com/whyrusleeping/cbor-gen/pull/50 should fix the issue. @whyrusleeping could you review it please?

**Review by magik6k — COMMENTED** [review]
This is API breaking, at the very least we should change the field name, maybe adding fallback to current behavior. Keeping it this way can cause people to make deals for way more FIL than they intended to

(interactive mode price math on [L815](https://github.com/filecoin-project/lotus/pull/5629/files#diff-2f58bf379b16a8226af249f73e0c214fc9b15bbe9af18d03ca84c734fa76e1e1R815) needs fixing)

**Comment by whyrusleeping** [comment]
@magik6k i can't imagine anyone actually uses an interface this broken for anything. The worst case is that someone pays 32 times more than they were expecting for a 32GB deal. I dont even think anyone has successfully made a 32gb deal using this interface.

One easy thing we can do is to put a little warning if the users price is above the miners listed ask price by 2x, and force them to add a flag if they really want to pay that much

**Comment by arajasek** [comment]
Let's extract out the helptext from this, but drop the (now-irrelevant) markets update

**Pull request closed without merging**